src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js

```diff
@@ -0,0 +1,236 @@
+import View from '@ckeditor/ckeditor5-ui/src/view';
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+import LabeledFieldView from '@ckeditor/ckeditor5-ui/src/labeledfield/labeledfieldview';
+
+import Model from '@ckeditor/ckeditor5-ui/src/model';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import { createLabeledInputText, createLabeledDropdown } from '@ckeditor/ckeditor5-ui/src/labeledfield/utils';
+import { addListToDropdown } from '@ckeditor/ckeditor5-ui/src/dropdown/utils';
+
+import { createLabeledInputNumber } from '../../common/input-number/utils';
+import { createLabeledSwitchButton } from '../../common/switch-button/utils';
+
+class IbexaCustomAttributesFormView extends View {
+    constructor(props) {
+        super(props);
+
+        this.locale = props.locale;
+
+        this.saveButtonView = this.createButton('Save custom attributes', null, 'ck-button-save', 'save-custom-attributes');
+        this.cancelButtonView = this.createButton('Remove custom attributes', null, 'ck-button-cancel', 'remove-custom-attributes');
+
+        this.attributeViews = {};
+        this.classesView = null;
+        this.attributeRenderMethods = {
+            string: this.createTextInput,
+            number: this.createNumberInput,
+            choice: this.createDropdown,
+            boolean: this.createBoolean,
+        };
+        this.setValueMethods = {
+            string: this.setStringValue,
+            number: this.setNumberValue,
+            choice: this.setChoiceValue,
+            boolean: this.setBooleanValue,
+        };
+    }
+
+    setChildren(customAttributes, customClasses) {
+        this.customAttributes = customAttributes;
+        this.customClasses = customClasses;
+        this.children = this.createFormChildren(customAttributes, customClasses);
+
+        this.setTemplate({
+            tag: 'form',
+            attributes: {
+                tabindex: '-1',
+            },
+            children: this.children,
+        });
+    }
+
+    setValues(attributesValues, classesValue) {
+        if (classesValue && this.classesView) {
+            this.setChoiceValue(this.classesView, classesValue);
+        }
+
+        Object.entries(attributesValues).forEach(([name, value]) => {
+            const attributeView = this.attributeViews[name];
+            const setValueMethod = this.setValueMethods[this.customAttributes[name].type];
+
+            if (!attributeView || !setValueMethod) {
+                return;
+            }
+
+            setValueMethod(attributeView, value);
+        });
+    }
+
+    getValues() {
+        const values = {
+            'custom-classes': this.classesView.fieldView.element.value,
+        };
+
+        Object.entries(this.attributeViews).forEach(([name, view]) => {
```

Comment:
could use `reduce` as well, but I'm not sure because of initial element in `values`...

```
const values = Object.entries(this.attributeViews).reduce((output, [name, view]) => {
  output[name] = view.fieldView.element.value;
}, { 'custom-classes': this.classesView.fieldView.element.value });
```